runhouse/resources/hardware/kubernetes_cluster.py

```diff
@@ -0,0 +1,127 @@
+import sky
+from pathlib import Path
+import subprocess
+import logging
+import time
+
+from runhouse.resources.hardware.on_demand_cluster import OnDemandCluster
+
+logger = logging.getLogger(__name__)
+
+class KubernetesCluster(OnDemandCluster):
+
+    def up(self):
+        """Up the cluster.
+
+        Example:
+            >>> rh.kubernetes_cluster(
+            >>>     name="cpu-cluster-test",
+            >>>     instance_type="CPU:1",
+            >>>     provider="kubernetes",      
+            >>> )
+        """
+        if self.on_this_cluster():
+            return self
+
+        if self.provider in ["kubernetes"]:
+            task = sky.Task(
```

Comment:
I think this section is the same as ondemandcluster's `up()`, we can extract the part inside the if statement to a helper function and reuse it directly to avoid duplicate code